src/main/java/com/wire/bots/roman/resources/BroadcastResource.java

```diff
@@ -206,6 +239,49 @@ private void broadcast(UUID providerId, UUID broadcastId, AudioPreview preview,
         }
     }
 
+    private void send(UUID providerId, UUID broadcastId, FileAssetPreview preview, FileAsset fileAsset, UUID botId) {
+        try {
+            if (fileAsset.getAssetKey() == null) {
+                AssetKey assetKey = sender.uploadAsset(fileAsset, botId);
+                if (assetKey != null) {
+                    fileAsset.setAssetToken(assetKey.token != null ? assetKey.token : "");
```

Comment:
I'd be cautious with this -> `fileAsset` is shared across all tasks in the thread pool ->  we should lock this operation if we want to upload this just once 

if we want to upload that multiple times, then we should copy this object

Reply:
with this implementation, there can be a race condition when Roman sends wrong decryption key to wrong asset

Reply:
oh. ofc. thanks